release.sh

```diff
@@ -3,8 +3,31 @@
 DIR=$(cd $(dirname ${BASH_SOURCE[0]}) && pwd)
 cd $DIR
 
-set -e
+git remote update
 
+remote=$(git rev-parse "@{u}") # '@{u}' gets the current upstream branch
+local=$(git rev-parse @) # '@' gets the current local branch 
+
+# check if local commit syncs with remote 
+if [ "$remote" != "$local" ]
+    then
+        echo "Error: local commit does not match remote. Exiting release script." 
+        exit 1
+fi 
+
+# remove any excess brackets, 'origin' branches and space/tab characters then sort the tags
+remoteTags=$(git ls-remote --tags origin | grep -v '{}' | sort | tr -d [[:blank:]])
+localTags=$(git show-ref --tags | grep -v '{}' | grep -v 'origin'| sort | tr -d [[:blank:]])
+
+# check if sorted local tags match remote tags
+if ! diff -s --left-column <(printf "%s" "$remoteTags") <(printf "%s" "$localTags") >/dev/null 2>&1
+    then 
+        echo "Error: local tags do not match remote. Exiting release script."
+        exit 1
+fi
+
+# cut the next Flux release
 version=$(go run github.com/influxdata/changelog nextver)
-git tag -s -m "Release $version" $version
+git tag -s -m "Release "$version"" "$version"
```

Comment:
What is this change for? It looks like it might pass three parameters instead of two to the call which would break it.

Reply:
Oh, that's my bad. I was trying to make sure that all my variables were wrapped in double quotes. I can change that back.

Reply:
Also, do we need the `set -e` line? It looks like it is supposed to quit out of the script if any command has a non-zero exit status. I added some error messages that might be helpful to the developer when they are working on debugging release issues. If it is necessary, I can add it back, but then I may as well remove the error messages and exit statuses as well.

Reply:
@nathanielc ^^

Reply:
I think we still want the `set -e` line as its provides some needed safety that commands are working as expected. The error messages will still be triggered when reached.